src/Microsoft.Identity.Abstractions/ApplicationOptions/CredentialDescriptionJsonConverter.cs

```diff
@@ -14,6 +15,8 @@ namespace Microsoft.Identity.Abstractions
     /// </summary>
     /// <remarks>BE CAREFUL when you serialize a credential description. The secrets it contains
     /// will be serialized too depending on the CredentialSource (ClientSecret, Base64Encoded, and Password)</remarks>
```

Comment:
We should expand the docs that for CustomSignedAssertionProviderData, JsonSerializerOptions.TypeInfoResolver can be used.